src/storage.ts

```diff
@@ -24,6 +24,22 @@ export interface CreateStorageOptions {
   driver?: Driver;
 }
 
+function filterKeyByDepth(key: string, depth: number | undefined): boolean {
+  if (depth === undefined) {
+    return true;
+  }
+
+  let substrCount = 0;
+  let index = key.indexOf(":");
+
+  while (index > -1) {
+    substrCount++;
+    index = key.indexOf(":", index + 1);
```

Comment:
We might perform a quick bench on this, my guess is single `split` / count would be faster since reduces userland/native roundtrips

Reply:
last time we benchmarked this algo in a parser I was working on, the order was `indexOf > iterate-by-character > split`

split was pretty slow in comparison

Reply:
Yep confirmed it is ~1.7x faster 👍🏼 

<summary>

```js
import assert from "node:assert";
import { summary, bench, run } from "mitata";

const impls = [
  {
    name: "indexOf (1)",
    fn: (key, depth) => {
      if (depth === undefined) {
        return true;
      }
      let substrCount = 0;
      let index = key.indexOf(":");
      while (index > -1) {
        substrCount++;
        index = key.indexOf(":", index + 1);
      }
      return substrCount <= depth;
    },
  },
  {
    name: "split",
    fn: (key, depth) => {
      return depth === undefined ? true : key.split(":").length <= depth + 1;
    },
  },
];

for (const impl of impls) {
  assert(impl.fn("a:b", 1) === true, `${impl.name} failed (true)`);
  assert(impl.fn("a:b:c", 1) === false, `${impl.name} failed (false)`);
}

summary(() => {
  for (const impl of impls) {
    bench(impl.name, () => {
      for (let i = 0; i < 1e5; i++) {
        impl.fn("a:b:c", 1);
        impl.fn("a:b", 1);
      }
    });
  }
});

await run();
```

</summary>`